### ISSUE TITLE
No GPU support for predictor

### ISSUE THREAD

**Issue by vivansxu** [nginyc/rafiki#141]
May I confirm if there is no GPU support for the predictor service?
If no, how do I implement prediction for models which have to use GPUs?
Is it possible to add GPU support for the predictor to Rafiki?
Besides, I notice it is said in the documentation that it is required for the model to train and evaluate with only CPUs if there is no GPU hardware available. Since my model only supports GPU environment, do I have to follow this requirement?
Thank you.

**Comment by nginyc** [comment]
Yes, there is currently no GPU support when the trained model is deployed for inference on Rafiki. Currently, it is expected that models can always fallback to using CPU. Is there a workaround for your model? @nudles should we support using GPU for inference as well?

**Comment by nudles** [comment]
Yes. It would be good to support GPU inference.
The implementation should be easy? similar to that for training.

**Comment by nginyc** [comment]
Shouldn't be that difficult. Will add as a task

**Comment by nginyc** [comment]
Hi @vivansxu, we've recently added this functionality on the branch [`add_gpu_for_inference`](https://github.com/nginyc/rafiki/tree/add_gpu_for_inference)! Let us know if it works for you. Refer to the [documentation on testing latest code changes](https://nginyc.github.io/rafiki/docs/0.2.0/src/dev/development.html#testing-latest-code-changes) to verify the changes since it's not on the `master` branch

**Comment by vivansxu** [comment]
Hi @nginyc, thank you so much for adding GPU support! I just tried to run the inference job, and it works well! 

By the way, since I want to return a list of strings as prediction, I changed line 64 of rafiki/predictor/ensemble.py from:
    if isinstance(prediction, Iterable):
to:
    if isinstance(prediction, list):

**Comment by nudles** [comment]
I think iterable is more general than list.
a list is also an Iterable instance.
If no other problems, I am going to merge the PR into the dev branch.

**Comment by vivansxu** [comment]
Hi @nudles, actually if just using Iterable here, since a string object is always iterable(explain at the end), there would be a RecursionError: maximum recursion depth exceeded in comparison

If I run:

for x in ('abc'):
    print(isinstance(x, Iterable))

The output would be:
True
True
True

**Comment by nginyc** [comment]
Hi @vivansxu 
Thanks for the catch. I have a fix for this bug in #132, which is pending to be merged into `dev`.
